### PR TITLE
tatello | stylus-config

### PR DESCRIPTION
**Pull request by tatelloGbnt**
**Goal:**
1.- Import a main.css class that contains the other styles of my app
2.-Devs can be debugger CSS environment and use just the main page for styles.

**Problem**
To create an app in CRA (create-react-app) library it has a config default and has this own Webpack configuration.
We need an external configuration to manipulate our files.

**Solution**
If you want config **Stylus** and their _source-map_, it should the config should be manual there are many ways to handle this solution, the best option is to create an external config with Webpack modules,  and config to able to handle source-map.
In package.json the configuration scripts were changed to handle the build .styl extension and their output folders.

In the end, we are able to import the main class to index.js and maintain a scalable style architecture.

image

Files to reference
Webpack: [Doc](https://webpack.js.org/concepts/modules/)

### CODE REVIEW

**Comment by tatelloGbnt** [comment]
my mistake close this PR

**Comment by jaconradog** [comment]
Could you please add some more context about what is happening here? Links to the documentation you used would be useful.

File: config-overrides.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+  webpack: (config, env) => {
+    if (env === 'development') {
+      config.module.rules.push({
+        test: /\.(js|mjs|jsx|ts|tsx)$/,
```
Comment (ddperaltam) [inline config-overrides.js]: creo que no es tan necesario toda la expresión, no está mal pero ¿en algún momento estamos usando typescript?

Comment (tatelloGbnt) [reply config-overrides.js]: you are right, sera mejor si acoto laas extensiones

**Review by gzepol2 — APPROVED** [review]
LGTM :)

**Pull request merged**